bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -237,10 +237,14 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     // Defer the creation of both the tableAdminClient until we need them.
     }
 
-  private ChannelPool getDataChannelPool() throws IOException {
+private ChannelPool getDataChannelPool() throws IOException {
     String host = options.getDataHost();
     int channelCount = options.getChannelCount();
-    if (options.useCachedChannel()) {
+    /**
+     * #1561-useCachedDataPool is now by default true hence for unit test cases to run successfully added null credential condition.
+     * As null credentials should not be used in production.  
+    */
+    if (options.useCachedChannel() && !CredentialOptions.nullCredential().equals(options.getCredentialOptions())) {
```

Comment:
What tests fail without this option?  It would probably be a good idea to fix those test cases.

Reply:
TestAppProfile is calling this method and as now we are setting options.useCachedChannel() value as true hence condition becomes true now & it is trying to load channelPool on localhost which is throwing following exception:
-----------------------------------------------------------------------------------------------------------------------------------------------------------
com.google.cloud.bigtable.grpc.async.AbstractRetryingOperation info
INFO: Retrying failed call. Failure #13, got: Status{code=UNAVAILABLE, description=io exception, cause=io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:37367
------------------------------------------------------------------------------------------------------------------------------------------------------------
In **TestAppProfile.java -> setup()** method I tried to set this **useCachedDataPool = false**  while building BigtableOptions to avoid above exception but this value will be overridden by our default value (true) and eventually above exception will come again.

Earlier as options.useCachedChannel() was false hence IF block never executed during test cases.

Hence to avoid exception I have added nullCredential() check because credentials will be null during test cases executions only.

Please let me know your thought on this.

Reply:
You've exposed a bug in `BigtableOptions` which is causing the problem in TestAppProfile.   I think that we should fix our Options classes. I'll create another bug for that.  We should fix `BigtableOptions` before merging this.

Reply:
BigtableOptions is fixed.  Please rebase, and see if ` !CredentialOptions.nullCredential().equals(options.getCredentialOptions())` can be removed